File: backend/src/main/java/outcobra/server/model/MarkReport.java
package outcobra.server.model;

import java.util.List;

import javax.persistence.*;
import javax.validation.constraints.NotNull;

import lombok.*;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class MarkReport {
    @Id
    private Long id;

    @NotNull
    private String name;

    @NotNull
    @ManyToOne
    private Semester semester;

    @OneToMany(mappedBy = "report")
    private List<MarkReportEntry> entries;
}

File: backend/src/main/kotlin/outcobra/server/model/MarkReport.kt
package outcobra.server.model

import outcobra.server.model.interfaces.ParentLinked
import javax.persistence.Entity
import javax.persistence.ManyToOne
import javax.persistence.OneToMany
import javax.validation.constraints.NotNull


@Entity
data class MarkReport(@NotNull var name: String = "",
                      @NotNull @ManyToOne var semester: Semester? = null,
                      @OneToMany(mappedBy = "report") var entries: List<MarkReportEntry> = arrayListOf())
    : ParentLinked, AbstractEntity() {
    override val parent: ParentLinked?
        get() = semester
}
